panther_manager/test/plugins/action/test_dock_robot_action_node.cpp

```diff
@@ -0,0 +1,247 @@
+{
+  CreateActionServer(
+    GoalResponse::ACCEPT_AND_EXECUTE, CancelResponse::ACCEPT, true, ActionResult::NONE);
+
+  std::map<std::string, std::string> params = {
+    {"action_name", "test_dock_action"},
+    {"use_dock_id", "true"},
+    {"dock_id", ""},
+    {"dock_type", "test_dock_type"},
+    {"max_staging_time", "5.0"},
+    {"navigate_to_staging_pose", "true"},
+  };
+
+  RegisterNodeWithParams<panther_manager::DockRobotAction>("DockRobotAction");
+  CreateTree("DockRobotAction", params);
+
+  auto & tree = GetTree();
+  auto status = tree.tickWhileRunning(std::chrono::milliseconds(1000));
+  EXPECT_EQ(status, BT::NodeStatus::FAILURE);
+}
+
+TEST_F(TestDockRobotAction, CallDockRobotActionServerWithEmptyDockType)
+{
+  CreateActionServer(
+    GoalResponse::ACCEPT_AND_EXECUTE, CancelResponse::ACCEPT, true, ActionResult::NONE);
+
+  std::map<std::string, std::string> params = {
+    {"action_name", "test_dock_action"},
+    {"use_dock_id", "true"},
+    {"dock_id", "main_dock"},
+    {"dock_type", ""},
+    {"max_staging_time", "5.0"},
+    {"navigate_to_staging_pose", "true"},
+  };
+
+  RegisterNodeWithParams<panther_manager::DockRobotAction>("DockRobotAction");
+  CreateTree("DockRobotAction", params);
+
+  auto & tree = GetTree();
+  auto status = tree.tickWhileRunning(std::chrono::milliseconds(1000));
+  EXPECT_EQ(status, BT::NodeStatus::FAILURE);
+}
+
+TEST_F(TestDockRobotAction, CallDockRobotActionServerWithNavigateToStagingPoseFailure)
+{
+  CreateActionServer(GoalResponse::REJECT, CancelResponse::ACCEPT, true, ActionResult::NONE);
+
+  std::map<std::string, std::string> params = {
+    {"action_name", "test_dock_action"},
+    {"use_dock_id", "true"},
+    {"dock_id", "main_dock"},
+    {"dock_type", "test_dock_type"},
+    {"max_staging_time", "5.0"},
+    {"navigate_to_staging_pose", "true"},
+  };
+
+  RegisterNodeWithParams<panther_manager::DockRobotAction>("DockRobotAction");
+  CreateTree("DockRobotAction", params);
+
+  auto & tree = GetTree();
+  auto status = tree.tickWhileRunning(std::chrono::milliseconds(1000));
+  EXPECT_EQ(status, BT::NodeStatus::FAILURE);
+}
+
+TEST_F(TestDockRobotAction, CallDockRobotActionServerWithNavigateToStagingPoseSuccess)
+{
+  CreateActionServer(
+    GoalResponse::ACCEPT_AND_EXECUTE, CancelResponse::ACCEPT, true, ActionResult::NONE);
+
+  std::map<std::string, std::string> params = {
+    {"action_name", "test_dock_action"},
+    {"use_dock_id", "true"},
+    {"dock_id", "main_dock"},
+    {"dock_type", "test_dock_type"},
+    {"max_staging_time", "5.0"},
+    {"navigate_to_staging_pose", "true"},
+  };
+
+  RegisterNodeWithParams<panther_manager::DockRobotAction>("DockRobotAction");
+  CreateTree("DockRobotAction", params);
+
+  auto & tree = GetTree();
+  auto status = tree.tickWhileRunning(std::chrono::milliseconds(1000));
+  EXPECT_EQ(status, BT::NodeStatus::SUCCESS);
+}
+
+int main(int argc, char ** argv)
+{
+  testing::InitGoogleTest(&argc, argv);
+
+  return RUN_ALL_TESTS();
+}
```

Comment:
Small detail, recently we decided to use a separate statement for result generation and return, so proposed approach is:

```suggestion
int main(int argc, char ** argv)
{
  testing::InitGoogleTest(&argc, argv);

  auto result = RUN_ALL_TESTS();

  return result;
}
```